src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,120 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoModal } from '../TodoModal';
+import cn from 'classnames';
 
-export const TodoList: React.FC = () => (
-  <table className="table is-narrow is-fullwidth">
-    <thead>
-      <tr>
-        <th>#</th>
-        <th>
-          <span className="icon">
-            <i className="fas fa-check" />
-          </span>
-        </th>
-        <th>Title</th>
-        <th> </th>
-      </tr>
-    </thead>
+type Props = {
+  todos: Todo[];
+  selectOption: string;
+  inputText: string;
+};
 
-    <tbody>
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-      <tr data-cy="todo" className="has-background-info-light">
-        <td className="is-vcentered">2</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = ({
+  todos,
+  selectOption,
+  inputText,
+}) => {
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
```

Comment:
I see it wasn't done
 
General rule - if we can define variable value based on other state values or props we don't need a separate state for it. Instead of updating it in useEffect create a common variable and wrap it in `useMemo`

```
const filteredTodos = useMemo(() => {
  // filtering logic and return filtered array
}, [// dependencies])
```

If you are not familiar with useMemo yet, you can omit useMemo part